3.4.1 build error on macOS: `DarwinProcess.c: error: ‘thread_extended_info_data_t’ undeclared`

```
    /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I.  -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -std=c99 -pedantic -D_DARWIN_C_SOURCE -DNCURSES_WIDECHAR -Wall -Wcast-align -Wcast-qual -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -DSYSCONFDIR="\"/opt/local/etc\"" -I"./darwin" -pipe -Os -arch ppc -MT zfs/ZfsArcMeter.o -MD -MP -MF $depbase.Tpo -c -o zfs/ZfsArcMeter.o zfs/ZfsArcMeter.c &&\
    mv -f $depbase.Tpo $depbase.Po
depbase=`echo zfs/ZfsCompressedArcMeter.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
    /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I.  -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -std=c99 -pedantic -D_DARWIN_C_SOURCE -DNCURSES_WIDECHAR -Wall -Wcast-align -Wcast-qual -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -DSYSCONFDIR="\"/opt/local/etc\"" -I"./darwin" -pipe -Os -arch ppc -MT zfs/ZfsCompressedArcMeter.o -MD -MP -MF $depbase.Tpo -c -o zfs/ZfsCompressedArcMeter.o zfs/ZfsCompressedArcMeter.c &&\
    mv -f $depbase.Tpo $depbase.Po
darwin/DarwinProcess.c: In function ‘DarwinProcess_scanThreads’:
darwin/DarwinProcess.c:427: warning: empty body in an if-statement
darwin/DarwinProcess.c:479: error: ‘thread_extended_info_data_t’ undeclared (first use in this function)
darwin/DarwinProcess.c:479: error: (Each undeclared identifier is reported only once
darwin/DarwinProcess.c:479: error: for each function it appears in.)
darwin/DarwinProcess.c:479: error: expected ‘;’ before ‘extended_info’
darwin/DarwinProcess.c:480: error: ‘THREAD_EXTENDED_INFO_COUNT’ undeclared (first use in this function)
darwin/DarwinProcess.c:481: error: ‘THREAD_EXTENDED_INFO’ undeclared (first use in this function)
darwin/DarwinProcess.c:481: error: ‘extended_info’ undeclared (first use in this function)
make[1]: *** [darwin/DarwinProcess.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_htop/htop/work/htop-3.4.1'
make: *** [all] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -186,6 +186,25 @@ AC_CHECK_HEADERS([execinfo.h])
 
 if test "$my_htop_platform" = darwin; then
    AC_CHECK_HEADERS([mach/mach_time.h])
+   AC_CHECK_HEADERS([mach/thread_info.h])
+
+   AC_MSG_CHECKING(for thread_extended_info_data_t)
+   AC_COMPILE_IFELSE(
+      AC_LANG_PROGRAM(
+         [#include <mach/thread_info.h>],
+         [
+            thread_extended_info_data_t teid;
+            (void)teid;
+         ]
+      ),
+      [
+         AC_DEFINE([HAVE_THREAD_EXTENDED_INFO_DATA_T], 1, [Define if thread_extended_info_data_t is available from <mach/thread_info.h>])
+         AC_MSG_RESULT(yes)
+      ],
+      [
+         AC_MSG_RESULT(no)
+      ]
+   )
 fi
 
 # ----------------------------------------------------------------------

--- a/darwin/DarwinProcess.c
+++ b/darwin/DarwinProcess.c
@@ -5,6 +5,8 @@ Released under the GNU GPLv2+, see the COPYING file
 in the source distribution for its full text.
 */
 
+#include "config.h" // IWYU pragma: keep
+
 #include "darwin/DarwinProcess.h"
 
 #include <libproc.h>
@@ -476,6 +478,7 @@ void DarwinProcess_scanThreads(DarwinProcess* dp, DarwinProcessTable* dpt) {
       tprocess->st_uid           = proc->st_uid;
       tprocess->user             = proc->user;
 
+#ifdef HAVE_THREAD_EXTENDED_INFO_DATA_T
       thread_extended_info_data_t extended_info;
       mach_msg_type_number_t extended_info_count = THREAD_EXTENDED_INFO_COUNT;
       ret = thread_info(thread_list[i], THREAD_EXTENDED_INFO, (thread_info_t) &extended_info, &extended_info_count);
@@ -495,9 +498,15 @@ void DarwinProcess_scanThreads(DarwinProcess* dp, DarwinProcessTable* dpt) {
          isProcessStuck |= true;
          tdproc->super.state = UNINTERRUPTIBLE_WAIT;
       }
+#endif
 
       // TODO: depend on setting
+#ifdef HAVE_THREAD_EXTENDED_INFO_DATA_T
       const char* name = extended_info.pth_name[0] != '\0' ? extended_info.pth_name : proc->procComm;
+#else
+      // Not provided in thread_basic_info_data_t; fall back to the process name
+      const char* name = proc->procComm;
+#endif
       Process_updateCmdline(tprocess, name, 0, name ? strlen(name) : 0);
 
       if (!preExisting)

